### PR TITLE
[Backport 2.x] Synchronize major changes from main 

### PR DESCRIPTION
**Pull request by peternied**
### Description
Manual backports have been failing for a while, specifically the failure on https://github.com/opensearch-project/security/pull/2926 clued me into the issue, we've been manually resolving these conflicts, but we should be trying to keep everything in sync as much as possible, this set of backports lumped together attempts to handle that.

### Issues Resolved
- Resolves https://github.com/opensearch-project/security/issues/3090

### Testing
Lots of `/gradlew testClasses`, CI checks should verify everything else.

### Leaving Draft Checklist
- [x] Check in on CI status 
- [x] ~GitHub workflows need to migrate?~ Not going to worry about these for the moment
- [x] Are other backports stuck waiting that should be pulled in?
- [x] Clean up commit history so they look like correctly executed backports
- [X] ~Note; this change should NOT be squashed merged, I'll manually push so commit history is preserved~ Actually its been paired down anyways while dealing with other conflicts, just going to squash.

### Check List
- [X] New functionality includes testing
- [X] New functionality has been documented
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -24,8 +26,12 @@ buildscript {
 
         common_utils_version = System.getProperty("common_utils.version", '2.9.0.0-SNAPSHOT')
         kafka_version  = '3.5.1'
-        jaxb_version = '2.3.8'
+        apache_cxf_version = '4.0.2'
+        open_saml_version = '4.3.0'
```
Comment (willyborankin) [inline build.gradle:30]: Could we we exclude so far OpenSAML 4.3.0 version?

Comment (cwperks) [reply build.gradle:30]: +1, OpenSAML has a known issue with permissions that needs to be resolved before backport to 2.x.

There's a PR that will resolve the permissions issue here: https://github.com/opensearch-project/security/pull/2991

Comment (peternied) [reply build.gradle:30]: Thanks for catching this

**Comment by peternied** [comment]
@cwperks @willyborankin Could I get another look?

**Comment by willyborankin** [comment]
> @cwperks @willyborankin Could I get another look?

LGTM.

**Review by willyborankin — APPROVED** [review]

File: build.gradle [hunk 2]
```diff
@@ -41,6 +47,8 @@ buildscript {
         maven { url "https://plugins.gradle.org/m2/" }
         maven { url "https://aws.oss.sonatype.org/content/repositories/snapshots" }
         maven { url "https://d1nvenhzbhpy0q.cloudfront.net/snapshots/lucene/" }
+        maven { url "https://build.shibboleth.net/nexus/content/groups/public" }
```
Comment (cwperks) [inline build.gradle:50]: The shibboleth repositories aren't needed after the reversion, but no harm in having them either.

File: src/main/java/com/amazon/dlic/auth/http/saml/AuthTokenProcessorHandler.java [hunk 3]
```diff
@@ -168,9 +168,7 @@ private AuthTokenProcessorAction.Response handleImpl(
 
         try {
 
-            SamlResponse samlResponse = new SamlResponse(saml2Settings, null);
```
Comment (cwperks) [inline src/main/java/com/amazon/dlic/auth/http/saml/AuthTokenProcessorHandler.java:171]: Is this change fine even after reverting back to the original version of opensaml?

**Pull request merged**